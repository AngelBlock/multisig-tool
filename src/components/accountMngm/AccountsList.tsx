import React, { FC, ReactElement } from "react";
import { List } from "antd";

interface Props {
  data: ({ accountId: string; amount: string } | undefined)[];
}

const AccountsList: FC<Props> = ({ data }): ReactElement => {
  return (
    <List
      header={<h3>Accounts list</h3>}
      bordered
      dataSource={data}
      renderItem={(item) => (
        <List.Item>
          <span data-test="account-item">{`${item?.accountId}: ${item?.amount}`}</span>
        </List.Item>
      )}
    />
  );
};

export default AccountsList;
