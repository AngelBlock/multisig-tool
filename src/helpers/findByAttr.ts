export default function findByAttr(component: any, attr: string): any {
  return component.find(`[data-test="${attr}"]`);
}
