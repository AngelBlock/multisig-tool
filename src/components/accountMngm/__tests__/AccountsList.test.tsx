import React from "react";
import { render } from "enzyme";
import findByAttr from "../../../helpers/findByAttr";
import AccountsList from "../AccountsList";

const mock = [
  {
    accountId: "test",
    amount: "123",
  },
];
describe("<AccountsList/>", () => {
  it("render properly", () => {
    const wrapper: any = render(<AccountsList data={mock} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("render props properly", () => {
    const wrapper: any = render(<AccountsList data={mock} />);
    const accountInfo = findByAttr(wrapper, "account-item");
    const shouldEqual = `${mock[0].accountId}: ${mock[0].amount}`;

    expect(accountInfo.text()).toEqual(shouldEqual);
  });
});
