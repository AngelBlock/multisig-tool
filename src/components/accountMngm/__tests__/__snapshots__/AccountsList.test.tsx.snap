// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountsList/> render properly 1`] = `
<div
  class="ant-list ant-list-split ant-list-bordered"
>
  <div
    class="ant-list-header"
  >
    <h3>
      Accounts list
    </h3>
  </div>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <span
            data-test="account-item"
          >
            test: 123
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
